faq.md

```diff
@@ -1,31 +1,51 @@
 ## FAQ questions
 
 ### Why use a new ABI? Why not an attribute?
+
 - Unlike ABIs, there is no syntax for assigning attributes to function
   pointers. The ABI must be part of the type system so that callers of function
   pointers know whether or not the function may unwind.
+
 ### What should I do if I want to have C++ functions that unwind into Rust? [DRAFT; TODO]
+
 - mark declarations in Rust `extern "C unwind"`
 - Do *not* use `catch_unwind` in calling code
 - compile w/ `panic = unwind`
 - Check compiler, linker, & platform documentation (TODO: expand on this)
 - Re-test for every compiler update
 - (TODO: other caveats?)
-### How does cross-language unwinding differ from cross-language
-  `setjmp`/`longjmp`?
-- `setjmp`/`longjmp` across Rust frames is currently guaranteed not to have
-  undefined behavior or to `abort`
-- It should never be assumed that `drop` will be called for objects in
-  intermediate frames traversed by a `longjmp`, but this may occur on certain
-  platforms. Rust provides no guarantee either way. Conversely, on platforms
-  where unwinding is supported, it will always call `drop`.
+
+### How does cross-language unwinding differ from cross-language `setjmp`/`longjmp`?
+
+- `setjmp`/`longjmp` across Rust frames is currently guaranteed to have
```

Comment:
do we have a citation for this? that's a strong term :) though I think it is reasonable.

Reply:
Since Rust itself cannot call them I don't think we have wording anywhere explicitly guaranteeing that this works. According to the requirements and guarantees we provide, when a Rust frame is "sandwiched" across C frames, and C does that, then as long as the frame has no types with destructors, none of the documented types of UB that we have in the reference can happen (https://github.com/rust-lang-nursery/reference/blob/master/src/behavior-considered-undefined.md).

We have gone to certain extends to make sure that this works for, e.g., rlua, making sure that longjmps of lua across Rust do not cause Rust to abort, even for `nounwind` functions.